### PR TITLE
Add support for yaml datetimes

### PR DESCRIPTION
**Pull request by Raymi306**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [ ] Have you created/updated the relevant documentation page(s)?

---

Issue #2071 
Please note that I wasn't fully sure how to incorporate my code with from_toml_datetime deserializer.
It looks like the time crate has [some sort of way to output rfc3339 strings](https://docs.rs/time/latest/time/format_description/well_known/index.html), and I'll try and wire that up if this PR is deemed acceptable.

I probably should write more tests if we want to keep this too.

I really ought to use verbose mode on the regex string, so I can add back in newlines and comments

### CODE REVIEW

**Comment by Keats** [comment]
Sorry I completely missed that. Are you still interested or should I finish it myself?

**Comment by Raymi306** [comment]
> Sorry I completely missed that. Are you still interested or should I finish it myself?

If you are interested in these changes, I should be able to clean up this PR and finish it.

> It looks like the time crate has [some sort of way to output rfc3339 strings](https://docs.rs/time/latest/time/format_description/well_known/index.html), and I'll try and wire that up if this PR is deemed acceptable.

I don't remember why I said this or if it was important.

**Comment by Keats** [comment]
Yep definitely go for it

File: components/utils/src/de.rs
Comment (Keats) [inline components/utils/src/de.rs:10]: Maybe the idea was to use the time crate to replace that?

**Comment by Raymi306** [comment]
I rebased and wrote some more tests

**Comment by Raymi306** [comment]
Some additional thoughts:
The spec I based my parsing off of was not finalized.
I used [this](https://yaml.org/type/timestamp.html) as a resource.
Since YAML is time format agnostic, it was valid to just specify that Zola only supported what it supports.

Jekyll is using something not quite the same as what was proposed for YAML:
> A date is specified in the format YYYY-MM-DD HH:MM:SS +/-TTTT; hours, minutes, seconds, and timezone offset are optional. 

This theoretically should be a simpler pattern to match, and would expand the supported time formats to be on par with Jekyll.
I guess it doesn't hurt to support Jekyll plus a little more which is what I ended up doing here.

File: components/content/src/front_matter/page.rs
Comment (Keats) [inline components/content/src/front_matter/page.rs:132]: this can be removed

**Review by Keats — APPROVED** [review]
LGTM, I'll remove the print after merging

**Comment by Keats** [comment]
Thanks!

**Pull request merged**
